tbs: updated to kernel-4.14

ck3d opened the pull request:
* updated meta data
* fix concurrent names for fetchFromGitHub derivations

###### Motivation for this change


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

ck3d committed:
tbs: updated to kernel-4.14

* updated meta data
* fix concurrent names for fetchFromGitHub derivations

cillianderoiste commented:
Hi @ck3d, your changes in #32049 were undone in 6a4cc3562cf6e41272e542f3aeb7ae46cec45ecc. Could you please do another PR, or change this one? Thanks!

ck3d committed:
Merge remote-tracking branch 'channels/nixpkgs-unstable' into tbs

ck3d commented:
Thanks, I updated this PR.

ck3d committed:
added lost entry in all-packages.nix

ck3d commented:
I updated the PR and added a missing entry in all-packages.nix. It got lost after the last update.

Sorry, but I can not test the packages any more. My DVB card seems to be broken. My system do not recognize the card any more.

dezgeg commented on pkgs/os-specific/linux/tbs/default.nix:
```diff
@@ -0,0 +1,58 @@
+  hardeningDisable = [ "pic" "format" ];
+
+  nativeBuildInputs = [ patchutils kmod perl perlPackages.ProcProcessTable ]
+  ++ lib.optional (lib.versionAtLeast kernel.version "4.14") [ libelf ];
```
Use `++ kernel.moduleBuildDependencies` instead of this conditional.

dezgeg commented on pkgs/os-specific/linux/tbs/default.nix:
```diff
@@ -0,0 +1,58 @@
+  buildFlags = [ "VER=${kernel.modDirVersion}" ];
+  installFlags = [ "DESTDIR=$(out)" ];
+
+  hardeningDisable = [ "pic" "format" ];
```
I think `hardeningDisable = [ "all" ];` is the safest thing to do with kernel modules.

cillianderoiste commented:
@ck3d I found that there's a conflict between the kernel module for dvb-core (dvb-core.ko.cz) and the one contained in this package (dvb-core.ko). The kernel version gets loaded, which is missing the dvb_create_media_graph symbol. I'm testing if xz compressing dvb-core.ko will be enough to fix this.

cillianderoiste committed:
tbs: override the kernel dvb-core.ko and videodev.ko

Otherwise the kernel versions of dvb-core.ko and videodev.ko get
loaded instead of the versions from this package.
